pkg/ccl/sqlproxyccl/proxy_handler.go

```diff
@@ -539,6 +545,36 @@ func (handler *proxyHandler) handle(ctx context.Context, incomingConn net.Conn)
 	}
 }
 
+// validateRequest validates the incoming connection by ensuring that the SQL
+// connection knows some additional information about the tenant (i.e. the
+// cluster name) before being allowed to connect.
+func (handler *proxyHandler) validateConnection(
+	ctx context.Context, tenantID roachpb.TenantID, clusterName string,
+) error {
+	tenant, err := handler.directoryCache.LookupTenant(ctx, tenantID)
+	if err != nil {
+		log.Errorf(ctx, "could not validate connection: %v", err.Error())
```

Comment:
nit: the "could not validate connection" logging statement isn't necessary since we are returning an error. IMO errors should only be logged at the point they are consumed. E.g. log the error when it is ignored or handled by closing the connection.

The "could not validate connection: cluster name '%s' doesn't match" is a useful log line because it does not contain information duplicated by the error.

Reply:
Done.

Reply:
Also discussed offline re: logging situation in the proxy.